test/unit/models/classes/Platform/Repository/Lti1p3RegistrationRepositoryTest.php

```diff
@@ -64,24 +69,28 @@ public function setUp(): void
             new Key('tool_public_key'),
             new Key('tool_private_key')
         );
+        $this->ltiProviderService = $this->createMock(LtiProviderService::class);
         $this->toolKeyChainRepository = $this->createMock(KeyChainRepositoryInterface::class);
-        $this->platformKeyChainRepository = $this->createMock(CachedPlatformKeyChainRepository::class);
+        $this->platformKeyChainRepository = $this->createMock(KeyChainRepositoryInterface::class);
```

Comment:
Why not?

Reply:
Because I am relying on a interface, therefore the interface should be mocked. Imagine you received this through the constructor with type hint `KeyChainRepositoryInterface`. You do not know nothing about the concrete class at this point.